src/core/proof.ml

```diff
@@ -60,45 +73,132 @@ type proof_state =
 (** Short synonym for qualified use. *)
 type t = proof_state
 
-(** [init builtins name a] returns an initial proof state for a theorem  named
+(** [init name a] returns an initial proof state for a theorem  named
     [name], which statement is represented by the type [a]. Builtin symbols of
-    [builtins] may be used by tactics, and have been declared. *)
+    [builtins] may be used by tactics, and have been declared.
+    The list of goals is not initialized *)
 let init : Pos.strloc -> term -> t = fun name a ->
   let proof_term = fresh_meta ~name:name.elt a 0 in
-  let proof_goals = [Goal.of_meta proof_term] in
-  {proof_name = name; proof_term; proof_goals}
+  {proof_name = name; proof_term; proof_goals = []}
+
+(** [typ_init m] returns a goal associated to the meta m *)
+let typ_init : meta -> Goal.t list = fun m ->
+  let goal_typ = Goal.GoalTyp (Goal.goal_typ_of_meta m) in
+  [goal_typ]
+
+(** [sort_init typ ter] returns a list of goals corresponding to the
+    typability of [typ] by a sort and checking eventually that term
+    [ter] has type [typ] *)
+let sort_init : term option -> term option -> Goal.t list * term =
+  fun typ ter ->
+  let goal_unif x = Goal.GoalUnif x in
+  let (typ,sort, to_solve) = match typ,ter with
+    | Some(typ),Some(ter) ->
+      let sort, to_solve2 = Infer.infer [] typ in
```

Comment:
style: `let (sort, to_solve2) = ...`. Idem later.

Reply:
fixed in ed44999